axis.go

```diff
@@ -407,14 +417,74 @@ func (DefaultTicks) Ticks(min, max float64) (ticks []Tick) {
 		if val >= min && val <= max && !found {
 			ticks = append(ticks, Tick{Value: val})
 		}
-		if math.Nextafter(val, val+minorDelta) == val {
-			break
-		}
 		val += minorDelta
 	}
 	return
 }
 
+func removeDuplicates(elements []float64) []float64 {
+	// Uses a map to record duplicates as we find them.
+	seen := make(map[float64]bool)
+	out := make([]float64,0,len(elements))
+
+	for _,v := range elements {
+		if seen[v] {
+			//No elements are added as they are duplicates.
+			continue
+		}
+			// Records this element as a seen element.
+			seen[v] = true
+			// Appends it to the result slice.
+			out = append(out, v)
+	}
+	// Returns the new slice.
+	return out
+}
+
+func findRightLevelOfPrecision(elements []float64) []float64 {
+
+	var result []float64
+
+	for i := 1; i < 308; i++ { //308 is the precision of the MaxFloat64 which we take as the maximum one here
+
+		var result []float64
+		var i10 float64
```

Comment:
Delete this and replace the following line with the gopher operator.

```
i10 := math.Pow10(i)
```

and remove the blank line following it.